mangaki/mangaki/views.py

```diff
@@ -891,6 +892,51 @@ def fix_suggestion(request, suggestion_id):
     return render(request, 'fix/fix_suggestion.html', context)
 
 
+def nsfw_grid(request):
+    #FIXME: remove duplicate suggestions OR group by work ID
+    nsfw_suggestion_list = Suggestion.objects.select_related(
+            'work', 'user', 'work__category').prefetch_related(
+            'work__category', 'evidence_set__user').filter(
+            problem__in=['nsfw', 'n_nsfw'], is_checked=False).exclude(
+            evidence__isnull=False
```

Comment:
The idea is to remove all entries on which the user already gave their opinion, so that the suggestion doesn't show up on the grid. I have to filter before the template in order for the pagination to be properly made – else some pages could be empty if I only hid those suggestions with conditions in the template.

However, I could still show those suggestions on the grid, but with a message "You already gave your opinion" (as I do when AJAX is on and the user clicks on a 'This is NSFW' button) if you prefer !

*I also had forgotten to remove evidences from other users when filtering : I've fixed that by using a `Q` object for queries !*